server/src/main/java/io/druid/segment/realtime/firehose/SqlFirehoseFactory.java

```diff
@@ -0,0 +1,173 @@
+      @JsonProperty("prefetchTriggerBytes") Long prefetchTriggerBytes,
+      @JsonProperty("fetchTimeout") Long fetchTimeout,
+      @JsonProperty("foldCase") Boolean foldCase,
+      @JsonProperty("database") SQLFirehoseDatabaseConnector sqlFirehoseDatabaseConnector,
+      @JacksonInject @Smile ObjectMapper objectMapper
+  )
+  {
+    super(
+        maxCacheCapacityBytes,
+        maxFetchCapacityBytes,
+        prefetchTriggerBytes,
+        fetchTimeout,
+        objectMapper
+    );
+    Preconditions.checkArgument(sqls.size() > 0, "No SQL queries provided");
+
+    this.sqls = sqls;
+    this.objectMapper = objectMapper;
+    this.sqlFirehoseDatabaseConnector = sqlFirehoseDatabaseConnector;
+    this.foldCase = (foldCase == null ? false : foldCase);
+    this.connectorConfig = null;
+  }
+
+  @Override
+  protected InputStream openObjectStream(String object, File fileName) throws IOException
+  {
+    Preconditions.checkNotNull(sqlFirehoseDatabaseConnector, "SQL Metadata Connector not configured!");
+    try (FileOutputStream fos = new FileOutputStream(fileName)) {
+      final JsonGenerator jg = objectMapper.getFactory().createGenerator(fos);
+      sqlFirehoseDatabaseConnector.retryWithHandle(
+          (handle) -> {
+            ResultIterator<Map<String, Object>> resultIterator = handle.createQuery(
+                object
+            ).map(
+                (index, r, ctx) -> {
+                  Map<String, Object> resultRow = foldCase ? new CaseFoldedMap() : new HashMap<>();
+                  ResultSetMetaData resultMetadata;
+                  try {
+                    resultMetadata = r.getMetaData();
+                  }
+                  catch (SQLException e) {
+                    throw new ResultSetException("Unable to obtain metadata from result set", e, ctx);
+                  }
+                  try {
+                    for (int i = 1; i <= resultMetadata.getColumnCount(); i++) {
+                      String key = resultMetadata.getColumnName(i);
+                      String alias = resultMetadata.getColumnLabel(i);
+                      Object value = r.getObject(i);
+                      resultRow.put(alias != null ? alias : key, value);
+                    }
+                  }
+                  catch (SQLException e) {
+                    throw new ResultSetException("Unable to access specific metadata from " +
+                                                 "result set metadata", e, ctx);
+                  }
+                  return resultRow;
+                }
+            ).iterator();
+            jg.writeStartArray();
+            while (resultIterator.hasNext()) {
+              jg.writeObject(resultIterator.next());
+            }
+            jg.writeEndArray();
+            jg.close();
+            return null;
+          },
+          (exception) -> {
+            final boolean isStatementException = exception instanceof StatementException ||
```

Comment:
Would you check this? https://github.com/druid-io/druid/pull/5441#discussion_r178673465

Reply:
sorry I missed that one.